### PR TITLE
ssh: add more options

### PR DESCRIPTION
**Pull request by zimbatm**

### CODE REVIEW

File: modules/programs/ssh.nix [hunk 1]
```diff
@@ -24,6 +28,15 @@ let
         description = "Specifies port number to connect on remote host.";
       };
 
+      forwardAgent = mkOption {
+        default = null;
+        type = types.nullOr types.bool;
```
Comment (rycee) [inline modules/programs/ssh.nix]: I think you can simply have `type = types.bool;` here and set the default to `false`. Then have

```nix
    ++ optional cf.forwardAgent "  ForwardAgent yes"
```
below.

File: modules/programs/ssh.nix [hunk 2]
```diff
@@ -118,12 +137,14 @@ let
...
     ["Host ${cf.host}"]
     ++ optional (cf.port != null)         "  Port ${toString cf.port}"
+    ++ optional (cf.forwardAgent != null) "  ForwardAgent ${yn cf.forwardAgent}"
     ++ optional cf.forwardX11             "  ForwardX11 yes"
     ++ optional cf.forwardX11Trusted      "  ForwardX11Trusted yes"
     ++ optional cf.identitiesOnly         "  IdentitiesOnly yes"
     ++ optional (cf.user != null)         "  User ${cf.user}"
     ++ optional (cf.identityFile != null) "  IdentityFile ${cf.identityFile}"
     ++ optional (cf.hostname != null)     "  HostName ${cf.hostname}"
+    ++ optional (length cf.sendEnv > 0)   "  SendEnv ${join " " cf.sendEnv}"
```
Comment (rycee) [inline modules/programs/ssh.nix]: Instead of `(length cf.sendEnv > 0)` use `cf.sendEnv != []`.

File: modules/programs/ssh.nix (hunk 1 above)
Comment (zimbatm) [reply modules/programs/ssh.nix]: Does it inherit the settings from the `*` host?

File: modules/programs/ssh.nix (hunk 2 above)
Comment (zimbatm) [reply modules/programs/ssh.nix]: fixed in 7e1f483

File: modules/programs/ssh.nix (hunk 1 above)
Comment (rycee) [reply modules/programs/ssh.nix]: That is a fair point! My suggestion would not allow setting a default value in the `*` block. Better to keep it as `types.nullOr types.bool`.

**Comment by rycee** [comment]
Squashed and rebased to master in 456e2d7ed546cf0d5c9b1a9cc01875d36411a3fa. Thanks!

**Pull request closed without merging**